diff --git a/setup.py b/setup.py
index ac9ba60..320b4e8 100644
--- a/setup.py
+++ b/setup.py
@@ -46,12 +46,13 @@ class CMakeBuild(build_ext):
             # All these are handled by pip
             f"-DCMAKE_LIBRARY_OUTPUT_DIRECTORY={extdir}{os.sep}",
             f"-DCUDNN_FRONTEND_KEEP_PYBINDS_IN_BINARY_DIR=OFF",
+            f"-DCUDNN_FRONTEND_USE_SYSTEM_DLPACK=ON",
+            f"-DCUDNN_FRONTEND_FETCH_PYBINDS_IN_CMAKE=OFF",
+    
         ]
 
         if is_windows:
-            cmake_args += [
-                f"-DCUDNN_FRONTEND_FETCH_PYBINDS_IN_CMAKE=ON",
-            ]
+            pass
         else:
             cmake_args += [
                 f"-DCUDNN_FRONTEND_FETCH_PYBINDS_IN_CMAKE=OFF",
