From dae7c2043806895642f2d19e24ca1133bf9c0db0 Mon Sep 17 00:00:00 2001
From: Andrew Howe <ahowe@anaconda.com>
Date: Fri, 7 Aug 2026 14:59:33 -0400
Subject: [PATCH 2/2] fix windows cuda library load

---
 include/cudnn_frontend_shim.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/cudnn_frontend_shim.h b/include/cudnn_frontend_shim.h
index d3391705..b6989dcc 100644
--- a/include/cudnn_frontend_shim.h
+++ b/include/cudnn_frontend_shim.h
@@ -72,7 +72,11 @@ load_cuda_so() {
     dlerror();
 
     // Attempt to open the cuda library
+    #ifdef _WIN32
+    HMODULE handle    = dlopen("nvcuda.dll", RTLD_NOW);
+    #else
     HMODULE handle    = dlopen("libcuda.so.1", RTLD_NOW);
+    #endif
     const char *error = reinterpret_cast<const char *>(dlerror());
     if (!handle || error) {
         // If opening the library fails, throw an exception with the error message
@@ -109,7 +113,11 @@ load_cudart_so() {
     }
 
     // List of potential libcudart libraries (Adding major version to support python package)
+    #ifdef _WIN32
+    constexpr const char *libs[] = {"cudart64_12.dll", "cudart64_13.dll"};
+    #else
     constexpr const char *libs[] = {"libcudart.so.12", "libcudart.so.13"};
+    #endif
     constexpr size_t num_libs    = sizeof(libs) / sizeof(libs[0]);
 
     HMODULE lib_handle = nullptr;
-- 
2.50.1 (Apple Git-155)

